cmds/core/uptime/uptime.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Get the time the machine has been up
+// Synopsis:
+//     uptime
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"time"
+)
+
+// loadavg takes in the contents of proc/loadavg,it then extracts and returns the three load averages as a string
+func loadavg(contents string) (loadaverage string, err error) {
+	loadavg := strings.Fields(contents)
+	if len(loadavg) < 3 {
+		return "", fmt.Errorf("invalid contents:the contents of proc/loadavg we are trying to process contain less than the required 3 loadavgs: %v", contents)
+	}
+	return loadavg[0] + "," + loadavg[1] + "," + loadavg[2], nil
+}
+
+// uptime takes in the contents of proc/uptime it then extracts and returns the uptime in the format Days , Hours , Minutes ,Seconds
+func uptime(contents string) (*time.Time, error) {
+	uptimeArray := strings.Fields(contents)
+	if len(uptimeArray) == 0 {
+		return nil, errors.New("the contents of proc/uptime we are trying to read are empty")
+	}
+	uptimeDuration, err := time.ParseDuration(string(uptimeArray[0]) + "s")
+	if err != nil {
+		return nil, errors.New("could not get uptime:error parsing the string uptime to a time duration")
```

Comment:
Don't discard the error here. usually its always a good idea to pass along the original error you received. time.ParseDuration returned an error, there's good information there, return it as part of this error using fmt.Errorf

Reply:
Done